Block Editor: Add new `TextAlignmentControl` component (#60841)
WordPress/gutenberg: packages/block-editor/src/components/text-alignment-control/README.md

# TextAlignmentControl

> The `TextAlignmentControl` component is responsible for rendering a control element that allows users to select and apply text align options to blocks or elements in the Gutenberg editor. It provides an intuitive interface for changing the text alignment by applying different alignments such as `left`, `center`, `right`.

Reviewer: ```suggestion
The `TextAlignmentControl` component is responsible for rendering a control that allows users to select and apply text alignment options to blocks or elements in the Gutenberg editor.
```

Nit: Small tweak to wording to fix the use of the verb `align` as a noun. The second sentence also doesn't add much and sort of overlaps the intent of the first sentence so I think we can remove it.

## Development guidelines

### Usage

> Renders the Text Align Component with `left`, `center`, `right` options.

Reviewer: ```suggestion
Render the Text Alignment Component with its default `left`, `center`, and `right` options.
```

```jsx
import { TextAlignmentControl } from '@wordpress/block-editor';

const MyTextAlignmentControlComponent = () => (
	<TextAlignmentControl
		value={ textTransform }
		onChange={ ( value ) => {
			setAttributes( { textTransform: value } );
		} }
	/>
);
```

### Props

The component accepts the following props.

#### value

The current value of the Text Alignment setting. Available options are `left|center|right|justify`.

-   Type: `String`

> -   Required: No

Reviewer: ```suggestion
The current value of the Text Alignment setting.

-   Type: `String`
-   Required: No
-   Options: `left`, `center`, `right`, `justify`
```

I'm not sure of the preferred approach here but the existing components use `Options` to list the possible values. I think we should be consistent here.

#### controls

> An array of strings for what controls to show, by default it shows all. Available options are `left|center|right`.

Reviewer 2: Is `justify` a valid option?

Reviewer: > Consumers can also opt-in to justify by overriding the controls prop.

This [comment](https://github.com/WordPress/gutenberg/pull/60841#issuecomment-2065709938) makes me think it is now and these docs just need updating to the latest.

Author: Updated by cc75116 and cfae31a

I've allowed the `justify` value and updated the documentation as well.

![image](https://github.com/WordPress/gutenberg/assets/54422211/44b13374-e93c-4a69-88a8-220ec98fa749)
